src/reducers/users.tsx

```diff
@@ -0,0 +1,109 @@
+import invariant from 'invariant';
+import { Reducer } from 'redux';
+import { ActionType, createAction, getType } from 'typesafe-actions';
+
+type UserId = number;
+
+export type User = {
+  id: UserId;
+  name: string;
+  email?: string;
+  permissions?: Array<string>;
+};
+
+export const actions = {
+  loadCurrentUser: createAction('LOAD_CURRENT_USER', (resolve) => {
+    return (payload: { user: User }) => resolve(payload);
+  }),
+  loadUser: createAction('LOAD_USER', (resolve) => {
+    return (payload: { user: User }) => resolve(payload);
+  }),
+  logOut: createAction('LOG_OUT', (resolve) => {
+    return () => resolve({});
+  }),
+};
+
+export type UsersState = {
+  currentUserId: UserId | null;
+  byId: { [id: number]: User };
+  byName: { [name: string]: UserId };
```

Comment:
I agree with @willdurand that we should only focus on log in / out for this patch. We probably won't ever need to load multiple users. I think you can remove `byName` and all related actions. We can always add them back in later if we need them.